runtime-modules/proposals/discussion/src/benchmarking.rs

```diff
@@ -0,0 +1,221 @@
+#![cfg(feature = "runtime-benchmarks")]
+use super::*;
+use crate::Module as ProposalsDiscussion;
+use core::convert::TryInto;
+use frame_benchmarking::{account, benchmarks};
+use membership::Module as Membership;
+use sp_std::prelude::*;
+use system as frame_system;
+use system::EventRecord;
+use system::Module as System;
+use system::RawOrigin;
+
+const SEED: u32 = 0;
+
+fn get_byte(num: u32, byte_number: u8) -> u8 {
+    ((num & (0xff << (8 * byte_number))) >> 8 * byte_number) as u8
+}
+
+// Method to generate a distintic valid handle
+// for a membership. For each index.
+// TODO: This will only work as long as max_handle_length >= 4
+fn handle_from_id<T: membership::Trait>(id: u32) -> Vec<u8> {
+    let min_handle_length = Membership::<T>::min_handle_length();
+    // If the index is ever different from u32 change this
+    let mut handle = vec![
+        get_byte(id, 0),
+        get_byte(id, 1),
+        get_byte(id, 2),
+        get_byte(id, 3),
+    ];
+
+    while handle.len() < (min_handle_length as usize) {
+        handle.push(0u8);
+    }
+
+    handle
+}
+
+fn assert_last_event<T: Trait>(generic_event: <T as Trait>::Event) {
+    let events = System::<T>::events();
+    let system_event: <T as frame_system::Trait>::Event = generic_event.into();
+    // compare to the last event record
+    let EventRecord { event, .. } = &events[events.len() - 1];
+    assert_eq!(event, &system_event);
+}
+
+fn member_account<T: membership::Trait>(
+    name: &'static str,
+    id: u32,
+) -> (T::AccountId, T::MemberId) {
+    let account_id = account::<T::AccountId>(name, id, SEED);
+    let handle = handle_from_id::<T>(id);
+
+    let authority_account = account::<T::AccountId>(name, 0, SEED);
+
+    Membership::<T>::set_screening_authority(RawOrigin::Root.into(), authority_account.clone())
+        .unwrap();
+
+    Membership::<T>::add_screened_member(
+        RawOrigin::Signed(authority_account.clone()).into(),
+        account_id.clone(),
+        Some(handle),
+        None,
+        None,
+    )
+    .unwrap();
+
+    (account_id, T::MemberId::from(id.try_into().unwrap()))
+}
+
+benchmarks! {
+    _ { }
+
+    add_post {
+        let i in 1 .. T::MaxWhiteListSize::get();
+
+        // TODO: this parameter doesn't affect the running time
+        // maybe we should bound it here with the UI limit?
+        let j in 0 .. 50000;
+
+        // We do this to ignore the id 0 because the `Test` runtime
+        // returns 0 as an invalid id but 0 as a valid one
+        let (_, _) = member_account::<T>("member", 0);
+        let (account_id, caller_member_id) = member_account::<T>("caller_member", 1);
+
+        let mut whitelisted_members = vec![caller_member_id];
+
```

Comment:
Why we use such a formula for id?

Reply:
Because in the mock implementation for `ActorOriginValidator` we have the id value 0 as invalid and the id value 1 as valid. So, I create a member that I ignorde with ID 0 and start from id 1 in the next line, then to make this loop easier I start from id 2.

Reply:
Ok. Please, add this as a comment.